beacon_node/network/src/network_beacon_processor/rpc_methods.rs

```diff
@@ -428,6 +429,105 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         Ok(())
     }
 
+    pub fn handle_light_client_updates_by_range(
+        self: &Arc<Self>,
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        substream_id: SubstreamId,
+        request_id: RequestId,
+        request: LightClientUpdatesByRangeRequest,
+    ) {
+        self.terminate_response_stream(
+            peer_id,
+            connection_id,
+            substream_id,
+            request_id,
+            self.clone()
+                .handle_light_client_updates_by_range_request_inner(
+                    peer_id,
+                    connection_id,
+                    substream_id,
+                    request_id,
+                    request,
+                ),
+            Response::LightClientUpdatesByRange,
+        );
+    }
+
+    /// Handle a `LightClientUpdatesByRange` request from the peer.
+    pub fn handle_light_client_updates_by_range_request_inner(
+        self: Arc<Self>,
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        substream_id: SubstreamId,
+        request_id: RequestId,
+        req: LightClientUpdatesByRangeRequest,
+    ) -> Result<(), (RpcErrorResponse, &'static str)> {
+        debug!(self.log, "Received LightClientUpdatesByRange Request";
+            "peer_id" => %peer_id,
+            "count" => req.count,
+            "start_period" => req.start_period,
+        );
+
+        // Should not send more than max light client updates
+        let max_request_size: u64 = req.max_requested();
+        if req.count > max_request_size {
+            return Err((
+                RpcErrorResponse::InvalidRequest,
+                "Request exceeded max size",
+            ));
+        }
+
+        let lc_updates = match self
+            .chain
+            .get_light_client_updates(req.start_period, req.count)
+        {
+            Ok(lc_updates) => lc_updates,
+            Err(e) => {
+                error!(self.log, "Unable to obtain light client updates";
+                    "request" => ?req,
+                    "peer" => %peer_id,
+                    "error" => ?e
+                );
+                return Err((RpcErrorResponse::ServerError, "Database error"));
+            }
+        };
+
+        for lc_update in lc_updates.iter() {
+            self.send_network_message(NetworkMessage::SendResponse {
+                peer_id,
+                response: Response::LightClientUpdatesByRange(Some(Arc::new(lc_update.clone()))),
+                request_id,
+                id: (connection_id, substream_id),
+            });
+        }
+
+        let lc_updates_sent = lc_updates.len();
+
+        if lc_updates_sent < req.count as usize {
+            debug!(
+                self.log,
+                "LightClientUpdatesByRange outgoing response processed";
+                "peer" => %peer_id,
+                "msg" => "Failed to return all requested light client updates. The peer may have requested data ahead of whats currently available.",
```

Comment:
I think we're still trying to avoid `msg` tags, as they collide with the main log message when logged in ELK.

```suggestion
                "info" => "Failed to return all requested light client updates. The peer may have requested data ahead of whats currently available",
```